dotnet/src/Connectors/Connectors.AI.OpenAI/ChatCompletion/AzureChatCompletion.cs

```diff
@@ -54,21 +53,21 @@ public sealed class AzureChatCompletion : AzureOpenAIClientBase, IChatCompletion
     }
 
     /// <inheritdoc/>
-    public Task<string> GenerateMessageAsync(
+    public Task<IReadOnlyList<IChatResult>> GetChatCompletionsAsync(
         ChatHistory chat,
-        ChatRequestSettings? requestSettings = null,
+        ChatRequestSettings? requestSettings,
         CancellationToken cancellationToken = default)
     {
         return this.InternalGenerateChatMessageAsync(chat, requestSettings ?? new(), cancellationToken);
     }
 
     /// <inheritdoc/>
-    public IAsyncEnumerable<string> GenerateMessageStreamAsync(
+    public IAsyncEnumerable<IChatStreamingResult> GetStreamingChatCompletionsAsync(
         ChatHistory chat,
         ChatRequestSettings? requestSettings = null,
         CancellationToken cancellationToken = default)
     {
-        return this.InternalGenerateChatMessageStreamAsync(chat, requestSettings ?? new(), cancellationToken);
+        return this.InternalGenerateStreamingChatMessageAsync(chat, requestSettings ?? new(), cancellationToken);
```

Comment:
The public methods here have a nullable ChatRequestSettings?, but the internal implementations require this arg to be non-null (hence the new() here)... if falling back to a new is valid, I'd rather see the value of requestSettings passed through here (null or not) to the Internal* methods.

Then, only those methods, where the value is actually used, can set it to new() or otherwise ignore it when it's null. Otherwise, we get different fallback behavior across different callers (some will inject new, others will Verify.NotNull() and error out, others may make something else up).